packages/earn-controller/src/EarnController.ts

```diff
@@ -0,0 +1,365 @@
+  }) {
+    super({
+      name: controllerName,
+      metadata: earnControllerMetadata,
+      messenger,
+      state: {
+        ...getDefaultEarnControllerState(),
+        ...state,
+      },
+    });
+
+    this.#initializeSDK();
+    this.refreshPooledStakingData().catch(console.error);
+
+    const { selectedNetworkClientId } = this.messagingSystem.call(
+      'NetworkController:getState',
+    );
+    this.#selectedNetworkClientId = selectedNetworkClientId;
+
+    this.messagingSystem.subscribe(
+      'NetworkController:stateChange',
+      (networkControllerState) => {
+        if (
+          networkControllerState.selectedNetworkClientId !==
+          this.#selectedNetworkClientId
+        ) {
+          this.#initializeSDK(networkControllerState.selectedNetworkClientId);
+          this.refreshPooledStakingData().catch(console.error);
+        }
+        this.#selectedNetworkClientId =
+          networkControllerState.selectedNetworkClientId;
+      },
+    );
+
+    // Listen for account changes
+    this.messagingSystem.subscribe(
+      'AccountsController:selectedAccountChange',
+      () => {
+        this.refreshPooledStakingData().catch(console.error);
+      },
+    );
+  }
+
+  #initializeSDK(networkClientId?: string) {
+    const { selectedNetworkClientId } = networkClientId
+      ? { selectedNetworkClientId: networkClientId }
+      : this.messagingSystem.call('NetworkController:getState');
+
+    const networkClient = this.messagingSystem.call(
+      'NetworkController:getNetworkClientById',
+      selectedNetworkClientId,
+    );
+
+    if (!networkClient?.provider) {
+      this.#stakeSDK = null;
+      return;
+    }
+
+    const provider = new Web3Provider(networkClient.provider);
+    const { chainId } = networkClient.configuration;
+
+    // Initialize appropriate contracts based on chainId
+    const config: StakeSdkConfig = {
+      chainId: convertHexToDecimal(chainId),
+    };
+
+    try {
+      this.#stakeSDK = StakeSdk.create(config);
+      this.#stakeSDK.pooledStakingContract.connectSignerOrProvider(provider);
+    } catch (error) {
+      this.#stakeSDK = null;
+      // Only log unexpected errors, not unsupported chain errors
+      if (
+        !(
+          error instanceof Error &&
+          error.message.includes('Unsupported chainId')
+        )
+      ) {
+        console.error('Stake SDK initialization failed:', error);
+      }
+    }
+  }
+
+  #getCurrentAccount() {
+    return this.messagingSystem.call('AccountsController:getSelectedAccount');
+  }
+
+  #getCurrentChainId(): number {
+    const { selectedNetworkClientId } = this.messagingSystem.call(
+      'NetworkController:getState',
+    );
+    const {
+      configuration: { chainId },
+    } = this.messagingSystem.call(
+      'NetworkController:getNetworkClientById',
+      selectedNetworkClientId,
+    );
+    return convertHexToDecimal(chainId);
+  }
+
+  async refreshPooledStakes(): Promise<void> {
```

Comment:
Nit: It might be nice to supply JSDoc for these methods so people know how they ought to be used, but I guess we can leave that for a future PR.